backends/pkg/reconciler/eventmesh/handler.go

```diff
@@ -32,63 +33,81 @@ func EventMeshHandler(ctx context.Context, listers Listers) func(w http.Response
-		}
-
-		fetchedEventTypes, err := listers.EventTypeLister.List(labels.Everything())
+		err = json.NewEncoder(w).Encode(eventMesh)
 		if err != nil {
-			logger.Errorw("Error listing eventTypes", "error", err)
+			logger.Errorw("Error encoding event mesh", "error", err)
+			http.Error(w, err.Error(), http.StatusInternalServerError)
 			return
 		}
+	}
+}
 
-		logger.Debugw("Fetched event types", "event types", fetchedEventTypes)
-
-		convertedEventTypeMap := make(EventTypeMap)
-		for _, et := range fetchedEventTypes {
-			namespaceEventTypeRef := NamespaceEventTypeRef(et)
+func BuildEventMesh(listers Listers, logger *zap.SugaredLogger) (error, EventMesh) {
+	err, convertedBrokers := fetchBrokers(listers.BrokerLister, logger)
+	if err != nil {
+		logger.Errorw("Error fetching and converting brokers", "error", err)
+		return err, EventMesh{}
+	}
 
-			if et.Spec.Reference != nil {
-				if br, ok := brokerMap[RefNameAndNamespace(et.Spec.Reference)]; ok {
-					// add to broker provided event types
-					// only add if it hasn't been added already
-					if !slices.Contains(br.ProvidedEventTypes, namespaceEventTypeRef) {
-						br.ProvidedEventTypes = append(br.ProvidedEventTypes, namespaceEventTypeRef)
-					}
-				}
-			}
+	brokerMap := make(map[string]*Broker)
+	for _, cbr := range convertedBrokers {
+		brokerMap[cbr.GetNameAndNamespace()] = cbr
+	}
 
-			if _, ok := convertedEventTypeMap[namespaceEventTypeRef]; ok {
-				logger.Debugw("Duplicate event type", "event type", namespaceEventTypeRef)
-				continue
-			}
+	fetchedEventTypes, err := listers.EventTypeLister.List(labels.Everything())
+	if err != nil {
+		logger.Errorw("Error listing eventTypes", "error", err)
+		return err, EventMesh{}
+	}
 
-			convertedEventType := convertEventType(et)
-			convertedEventTypeMap[namespaceEventTypeRef] = &convertedEventType
+	sort.Slice(fetchedEventTypes, func(i, j int) bool {
+		if fetchedEventTypes[i].Namespace != fetchedEventTypes[j].Namespace {
+			return fetchedEventTypes[i].Namespace < fetchedEventTypes[j].Namespace
 		}
+		return fetchedEventTypes[i].Name < fetchedEventTypes[j].Name
+	})
 
-		eventMesh := EventMesh{
-			EventTypes: make([]*EventType, 0, len(convertedEventTypeMap)),
-			Brokers:    convertedBrokers,
-		}
+	logger.Debugw("Fetched event types", "event types", fetchedEventTypes)
 
-		for _, et := range convertedEventTypeMap {
-			eventMesh.EventTypes = append(eventMesh.EventTypes, et)
+	convertedEventTypeMap := make(EventTypeMap)
+	for _, et := range fetchedEventTypes {
+		namespaceEventTypeRef := NamespaceEventTypeRef(et)
+
+		if et.Spec.Reference != nil {
+			if br, ok := brokerMap[RefNameAndNamespace(et.Spec.Reference)]; ok {
+				// add to broker provided event types
+				// only add if it hasn't been added already
+				if !slices.Contains(br.ProvidedEventTypes, namespaceEventTypeRef) {
+					br.ProvidedEventTypes = append(br.ProvidedEventTypes, namespaceEventTypeRef)
+				}
+			}
 		}
 
```

Comment:
The reference can be any resource in v1beta2, not only brokers, so this leaves out sources from the catalog

Reply:
Yes, that's fine for now.
Currently the plugin will only show brokers and event types